src/ui/handlers/starter-select-ui-handler.ts

```diff
@@ -3199,6 +3270,7 @@ export class StarterSelectUiHandler extends MessageUiHandler {
     this.dexAttrCursor = 0n;
```

Comment:
```suggestion
    const oldTeraType = this.teraCursor !== ElementalType.UNKNOWN ? this.teraCursor : species?.type1 ?? ElementalType.UNKNOWN;
    this.dexAttrCursor = 0n;
```
(Fixing a bug where cycling nature, ability, etc. resets the selected Tera type.)